dspace/config/spring/api/discovery.xml

```diff
@@ -306,7 +474,7 @@
         <property name="defaultFilterQueries">
             <list>
                 <!--Only find items-->
-                <value>search.resourcetype:Item</value>
+                <value>search.resourcetype:Item AND latestVersion:true</value>
```

Comment:
I'm a bit hesitant here. We are forcing users to add this filter to all their configuration but this is mostly a business requirement than a configuration. Would it better to deal with it similarly to what we do with the withdrawn or discoverable flag? as a parameter in the SolrService methods and/or a SearchIndexPlugin?

Reply:
Summary of what we discussed in the meeting:
* I was also originally looking for a good solution, and did weigh a few options
* Because this PR had to deal with many unrelated/unexpected issues, we didn't want a larger change to discovery as well
* The hardcoded solution similar to the workflow doesn't work here because it's not a single configuration (you can have as many as you want, and define them how you'd want)
* The alternative I had in mind was a special DiscoveryConfiguration for this purpose, which doesn't filter on `latestVersion`. The default DiscoveryConfiguration would filter on `latestVersion`. Although this is possible, I would want to keep the scope still limited since this is possible via configuration

The change does indeed require discovery re-indexing. When 7.3 is released, the many DB changes will of course already start a discovery re-index